ext/pdo/pdo_stmt.c

```diff
@@ -1365,6 +1365,11 @@ PHP_METHOD(PDOStatement, fetchAll)
 	}
 
 	PDO_STMT_CLEAR_ERR();
+	/* Increase refcount for ctor_args as those might be removed during individual fetches */
+	bool increase_refcount_ctor = Z_TYPE(stmt->fetch.cls.ctor_args) == IS_ARRAY;
+	if (increase_refcount_ctor) {
```

Comment:
I wonder if it's better to combine this with the ZVAL_COPY_VALUE, i.e. make that ZVAL_COPY. However, this is not hugely important (it also depends on how this would complicate the resolution of my second comment).